woocommerce/woocommerce
# Update shipping partner suggestions to use data from API

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Closes #38456.

Adds consumption of shipping partner suggestions from WCCOM API.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

**Testing row layout**
1. Go to `WooCommerce > Settings > Advanced > WooCommerce.com` (/wp-admin/admin.php?page=wc-settings&tab=advanced&section=woocommerce_com)
1. Uncheck `Display suggestions within WooCommerce` option and save
1. Go to Settings > General and choose `Italy` country and save
1. Delete all shipping zone in Settings > Shipping and delete the option `woocommerce_admin_created_default_shipping_zones`. 
3. Go to WooCommerce > Home and click on `Add shipping costs`
4. Click on `Save shipping options`
5. Observe SendCloud and Packlink are shown

<img width="762" alt="image" src="https://github.com/woocommerce/woocommerce/assets/3747241/5d149026-afbc-40dd-9b25-198112ad5133">


**Testing column layout**
1. Go to Settings > General and choose `Canada` country and save
1. Delete all shipping zone in Settings > Shipping and delete the option `woocommerce_admin_created_default_shipping_zones`. 
3. Go to WooCommerce > Home and click on `Add shipping costs`
4. Click on `Save shipping options`
5. Observe that ShipStation is shown

<img width="730" alt="image" src="https://github.com/woocommerce/woocommerce/assets/3747241/c829fa35-445b-44bb-8c86-26b539bf687f">


**Optional further testing**
7. Optionally, repeat testing for any countries from this list `'CA', 'AU', 'GB', 'ES', 'IT', 'DE', 'FR', 'MX', 'CO', 'CL', 'AR', 'PE', 'BR', 'UY', 'GT', 'NL', 'AT', 'BE'` and ensure it follows the rules in [DefaultShippingPartners.php](https://github.com/woocommerce/woocommerce/blob/d850244ad63c1db5dd970c3e5eb08c8a5f2ab81e/plugins/woocommerce/src/Admin/Features/ShippingPartnerSuggestions/DefaultShippingPartners.php)
